pkgs/firehose/lib/src/health/health.dart

```diff
@@ -56,35 +71,48 @@ class Health {
       return;
     }
 
-    print('Start health check for the checks $args');
-    var checks = [
-      if (args.contains('version') &&
-          !github.prLabels.contains('skip-validate-check'))
-        validateCheck,
-      if (args.contains('license') &&
-          !github.prLabels.contains('skip-license-check'))
-        licenseCheck,
-      if (args.contains('changelog') &&
-          !github.prLabels.contains('skip-changelog-check'))
-        changelogCheck,
-      if (args.contains('coverage') &&
-          !github.prLabels.contains('skip-coverage-check'))
-        (GithubApi github) => coverageCheck(github, coverageweb),
-      if (args.contains('breaking') &&
-          !github.prLabels.contains('skip-breaking-check'))
-        breakingCheck,
-      if (args.contains('do-not-submit') &&
-          !github.prLabels.contains('skip-do-not-submit-check'))
-        doNotSubmitCheck,
-    ];
-    final results = <HealthCheckResult>[];
-    for (var check in checks) {
-      results.add(await check(github));
+    print('Start health check for the check $check');
+    print('Checking for $check');
+    if (!github.prLabels.contains('skip-$check-check')) {
+      final firstResult = await checkFor(check)();
+      final HealthCheckResult finalResult;
+      if (warnOn.contains(check) && firstResult.severity == Severity.error) {
+        finalResult = firstResult.withSeverity(Severity.warning);
+      } else if (failOn.contains(check) &&
+          firstResult.severity == Severity.warning) {
+        finalResult = firstResult.withSeverity(Severity.error);
+      } else {
+        finalResult = firstResult;
+      }
+      await writeInComment(github, finalResult);
+      print('\n\n${finalResult.severity.name.toUpperCase()}: $check done.\n\n');
+    } else {
+      print('Skipping $check, as the skip tag is present.');
     }
-    await writeInComment(github, results);
   }
 
-  Future<HealthCheckResult> validateCheck(GithubApi github) async {
+  String tagFor(String checkType) => switch (checkType) {
```

Comment:
Nit: these two switches feel like you could introduce a `class CheckType` with fields `String  tag` and `Future<HealthCheckResult> Function() check`.

`checkTypes` could be `CheckType.values.map((c)=>c.tag)` etc.